app/src/main/java/org/p2p/wallet/bridge/interactor/EthereumInteractor.kt

```diff
@@ -0,0 +1,69 @@
+package org.p2p.wallet.bridge.interactor
+
+import java.math.BigInteger
+import org.p2p.core.token.Token
+import org.p2p.core.wrapper.HexString
+import org.p2p.core.wrapper.eth.EthAddress
+import org.p2p.ethereumkit.external.repository.EthereumRepository
+import org.p2p.ethereumkit.internal.models.Signature
+import org.p2p.wallet.bridge.claim.interactor.ClaimInteractor
+import org.p2p.wallet.bridge.claim.model.ClaimStatus
+import org.p2p.wallet.bridge.model.BridgeBundle
+import org.p2p.wallet.bridge.send.BridgeSendInteractor
+
+class EthereumInteractor(
+    private val claimInteractor: ClaimInteractor,
+    private val sendInteractor: BridgeSendInteractor,
+    private val ethereumRepository: EthereumRepository,
+) {
+
+    fun setup(userSeedPhrase: List<String>) {
+        ethereumRepository.init(userSeedPhrase)
+    }
+
+    suspend fun loadWalletTokens(): List<Token.Eth> {
+        return ethereumRepository.loadWalletTokens()
+    }
+
+    suspend fun getEthereumToken(): Token.Eth? {
+        return ethereumRepository.getUserEthToken()
+    }
+
+    suspend fun getEthAddress(): EthAddress {
+        return ethereumRepository.getAddress()
+    }
+
+    suspend fun getEthereumBundle(erc20Token: EthAddress?, amount: String): BridgeBundle {
+        val ethereumAddress: EthAddress = ethereumRepository.getAddress()
+        return claimInteractor.getEthereumBundle(
+            erc20Token = erc20Token,
+            amount = amount,
+            ethereumAddress = ethereumAddress
+        )
+    }
+
+    suspend fun getListOfEthereumBundleStatuses(): Map<String, List<ClaimStatus>> {
```

Comment:
Maybe you can type String to Base58String or HexString?

Reply:
yeah, I can, but if I will do this, I need to override String type of Token.Active, Token.Eth classes.
So it's will be a huge bunch of changes x D

Reply:
or I can wrap and unwrap x D

Reply:
no I can not x D